magma/clock.py

```diff
@@ -19,6 +20,16 @@ def unused(self):
     def undriven(self):
         Bit.undriven(self)
 
+    @debug_wire
```

Comment:
this kind of an interesting pattern. if we think of the inheritance as a tree, the default is for any nodes of the tree (types) to be wireable if they share a common ancestor, but sometimes we want to break that and say two types have to have a specific common ancestor, and we basically do that by overriding `def wire()` to do this check. The interesting thing is that if we wanted apply the same thing to other types the code would be identical, but we can't get reuse through inheritance b/c that's the problem in the first place!

Just an observation.